### ISSUE TITLE
"Bundler clean" do not follow symlink for Git gems

### ISSUE THREAD

**Issue by Haelle** [bundler/bundler#7378]
### Situation : 
- We use Mina for deployment 
- In the Gemfile we have a gem referenced with `gem 'foo', git: 'https://...'`

### Problem
We wanted to use the option `--clean` but we encountered an issue : 
https://github.com/bundler/bundler/blob/f75efc27501ca4390abbae3d8846d0e54dab048b/lib/bundler/runtime.rb#L186
when we evaluate these vars we have : 
`git_dirs` : /path/to/mina/project/shared/vendor/../gems/foo
`spec_git_path` : /path/to/mina/project/release/XX/vendor/.../gems/foo
which are the same directories, the second one is a symlink to the first one ! This is how Mina handle deployments.

### Proposition
so `spec_git_path` should evaluate to the real path right ?
It would need to change : 
https://github.com/bundler/bundler/blob/f75efc27501ca4390abbae3d8846d0e54dab048b/lib/bundler/definition.rb#L319-L321
to this: 
```ruby
def spec_git_paths
  sources.git_sources.map {|s| s.path.realpath }
end
```

I can open a PR but I have no idea how to add a valid spec

**Comment by deivid-rodriguez** [comment]
This should've been fixed by https://github.com/bundler/bundler/pull/7211 and https://github.com/bundler/bundler/pull/7340. Can you try bundler 2.1.0.pre.2 and confirm that your issue is fixed?

**Comment by Haelle** [comment]
I confirm this is fixed in 2.1.0.pre.2.
Sorry for the duplicate, I looked in the issues but not in the PRs...

**Comment by deivid-rodriguez** [comment]
No worries! :+1:
